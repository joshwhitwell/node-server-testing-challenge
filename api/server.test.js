const request = require('supertest')
const server = require('./server')

describe('server.js', () => {
    it('uses testing environment', () => {
        expect(process.env.DB_ENV).not.toBe('development')
        expect(process.env.DB_ENV).not.toBe('production')
        expect(process.env.DB_ENV).toBe('testing')
    })

    describe('[GET] /', () => {
        it('returns correct data', async () => {
            const res = await request(server).get('/')
            expect(res.status).toBe(200)
            expect(res.body).toMatchObject({ api: 'up' })
            expect(res.type).toMatch(/json/)
            expect(res.body).toMatchSnapshot()
        })
    })
})