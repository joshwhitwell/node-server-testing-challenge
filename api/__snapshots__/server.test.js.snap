// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`server.js [GET] / returns correct data 1`] = `
Object {
  "api": "up",
}
`;
